modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/FilteringVectorProcess.java

```diff
@@ -0,0 +1,423 @@
+            }
+            Operations op = Operations.valueOf(aggregation);
+            FeatureType schema = features.getSchema();
+            NamespaceSupport ns = declareNamespaces(schema);
+            List<PropertyName> groupingPn =
+                    groupingAttributes
+                            .stream()
+                            .map(
+                                    g ->
+                                            validatePropertyName(
+                                                    new AttributeExpressionImpl(g, ns), schema))
+                            .collect(Collectors.toList());
+            PropertyName opValue =
+                    validatePropertyName(ff.property(operationAttribute, ns), schema);
+            return new GroupingFeatureCollection(features, groupingPn, opValue, op);
+        } catch (IllegalArgumentException e) {
+            throw new ProcessException(
+                    Errors.format(ErrorKeys.BAD_PARAMETER_$2, "aggregation", aggregation));
+        }
+    }
+
+    public Query invertQuery(
+            @DescribeParameter(
+                        name = "operationAttribute",
+                        description =
+                                "The feature's attribute to be used to compute the aggregation",
+                        min = 1
+                    )
+                    String operationAttribute,
+            @DescribeParameter(
+                        name = "groupingAttributes",
+                        description =
+                                "The feature's attributes defining groups for which perform the filtering based on the aggregation operation and the operation attribute."
+                                        + "Consistent results are guaranteed only if the vector process is fed with features already sorted  by these attributes",
+                        min = 1
+                    )
+                    List<String> groupingAttributes,
+            Query targetQuery,
+            GridGeometry gridGeometry) {
+
+        // optimize the query adding the needed property name to it
+
+        List<PropertyName> properties = targetQuery.getProperties();
+        SortBy[] sorts = targetQuery.getSortBy();
+
+        Query q = targetQuery != null ? new Query(targetQuery) : new Query();
+        // add the sortBy if not present
+        SortBy[] sortBy = buildNewSortBy(sorts, groupingAttributes);
+        q.setSortBy(sortBy);
+
+        // produces new PropertyName to add to the query
+        List<PropertyName> propertiesToAdd =
+                Stream.of(sortBy).map(s -> s.getPropertyName()).collect(Collectors.toList());
+        propertiesToAdd.add(ff.property(operationAttribute));
+
+        // eventually merge with existing ones
+        List<PropertyName> pns = getNewProperties(propertiesToAdd, properties);
+        q.setProperties(pns);
+
+        // features with null values for groupingAttributes and operationAttribute
```

Comment:
A database in general can group over nulls, it's just a different value. IMHO this should not be hard-coded, but part of the filtering done in the SLD. If this is a hard requirement, it needs to be clearly documented, as it goes against normal practice (TBH, I would consider this behavior a bug).

For reference: https://learnsql.com/blog/null-values-group-clause/

Reply:
fixed